src/providers/mod.rs

```diff
@@ -26,32 +26,47 @@ pub mod mbed_provider;
 #[cfg(feature = "tpm-provider")]
 pub mod tpm_provider;
 
-#[derive(Copy, Clone, Deserialize, Debug)]
-pub enum ProviderType {
-    MbedProvider,
-    Pkcs11Provider,
-    TpmProvider,
+#[derive(Deserialize, Debug)]
+#[serde(tag = "provider_type")]
```

Comment:
I don't understand what this `serde` attribute is for 😕

Reply:
We need that for correctly parsing the parsec config file into the enum of structs. See "Internally tagged" on this page:
https://serde.rs/enum-representations.html

I though about adding a comment with the link, but it's the first link in google search results for "serde tag". I can add it if useful.

Reply:
I don't think it needs a comment, but @hug-dev might disagree. 

As for the feature use itself, I think it's very appropriate, makes it a lot more explicit what you need for each Provider. Thanks! :)

Reply:
Oh I see, with this tag the deserialization into the enum of structs maps exactly our `provider` format in the config file, because we already had the `provider_type` field. That way we do not have to change the configuration, that is neat!

I always think that a one line comment never hurt, even if it is superfluous. But only do it if you need to make other changes to the PR 😃

Reply:
done